src/main/kotlin/com/github/suusan2go/kotlinfillclass/intentions/FillClassIntention.kt

```diff
@@ -1,57 +1,57 @@
-        return text
+class FillClassIntention : SelfTargetingIntention<KtValueArgumentList>(KtValueArgumentList::class.java, "Fill class constructor") {
+    override fun isApplicableTo(element: KtValueArgumentList, caretOffset: Int): Boolean {
+        val parameters = element.getValueParameters() ?: return false
+        return element.arguments.size != parameters.size
     }
 
-    override fun getText(): String = "Fill class constructor"
-
-    override fun invoke(project: Project, editor: Editor?, file: KtFile) {
-        val analysisResult = element?.calleeExpression?.analyzeAndGetResult() ?: return
-        val classDescriptor = element
-                ?.calleeExpression
-                ?.getReferenceTargets(analysisResult.bindingContext)
-                ?.mapNotNull { (it as? ConstructorDescriptor)?.containingDeclaration }
-                ?.distinct()
-                ?.singleOrNull() ?: return
-        val parameters = classDescriptor.constructors.first().valueParameters
+    override fun applyTo(element: KtValueArgumentList, editor: Editor?) {
+        val parameters = element.getValueParameters() ?: return
+        createParameterSetterExpression(element, parameters)
+    }
 
-        val factory = KtPsiFactory(project = project)
-        val argument = factory.createExpression("""${classDescriptor.name.identifier}(
-            ${createParameterSetterExpression(parameters)}
-            )""".trimMargin())
-        element?.replace(argument)
-        return
+    private fun KtValueArgumentList.getValueParameters(): MutableList<ValueParameterDescriptor>? {
+        val callExpression = getStrictParentOfType<KtCallExpression>() ?: return null
+        val calleeExpression = callExpression.calleeExpression ?: return null
+        val analysisResult = calleeExpression.analyzeAndGetResult()
+        val classDescriptor = calleeExpression
+                .getReferenceTargets(analysisResult.bindingContext)
+                .mapNotNull { (it as? ConstructorDescriptor)?.containingDeclaration }
+                .distinct()
+                .singleOrNull() ?: return null
+        return classDescriptor.constructors.first().valueParameters
     }
 
     override fun startInWriteAction() = true
 
-    private fun createParameterSetterExpression(parameters: List<ValueParameterDescriptor>): String {
-        var result = ""
-        parameters.forEach { parameter ->
-            var parameterString = "${parameter.name.identifier} = ${createDefaultValueFromParameter(parameter)},\n".let {
-                if(parameters.last() == parameter) return@let it.replace(",\n","")
-                it
-            }
-            result = "$result$parameterString"
+    private fun createParameterSetterExpression(element: KtValueArgumentList, parameters: List<ValueParameterDescriptor>) {
+        val arguments = element.arguments
+        val argumentNames = arguments.map { it.getArgumentName()?.asName?.identifier }.filterNotNull()
```

Comment:
I think we can simplify this by `mapNotNull` 👍

Reply:
Good catch!